lib/sharingpluginloader.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2021 Open Mobile Platform LLC.
+ *
+ * All rights reserved.
+ *
+ * This file is part of Sailfish Transfer Engine package.
+ *
+ * You may use this file under the terms of the GNU Lesser General
+ * Public License version 2.1 as published by the Free Software Foundation
+ * and appearing in the file license.lgpl included in the packaging
+ * of this file.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1 as published by the Free Software Foundation
+ * and appearing in the file license.lgpl included in the packaging
+ * of this file.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ */
+
+#include <QDebug>
+#include <QDir>
+#include <QPluginLoader>
+#include <QString>
+
+#include "sharingplugininfo.h"
+#include "sharingplugininterface.h"
+#include "sharingpluginloader_p.h"
+#include "sharingpluginloader.h"
+
+namespace {
+const int FileWatcherTimeout = 5000;
+const auto SharePluginsPath = QStringLiteral(SHARE_PLUGINS_PATH);
+} // namespace
+
+SharingPluginLoaderPrivate::SharingPluginLoaderPrivate(SharingPluginLoader *parent)
+    : QObject(parent)
+    , q_ptr(parent)
+    , m_loading(true)
+    , m_accountManager("sharing")
```

Comment:
Wonder if we could get rid of this. Now seems to trigger only plugin change signal which external code can use to trigger reloading. Practically it's quite hard now to get new accounts created while a sharing method list instance is alive. On API it would be also good if the plugins had some way to note themselves they have changed information.